hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceWALReader.java

```diff
@@ -178,6 +183,7 @@ protected final boolean addEntryToBatch(WALEntryBatch batch, Entry entry) {
     }
     LOG.trace("updating TimeStampOfLastAttempted to {}, from entry {}, for source queue: {}",
       entry.getKey().getWriteTime(), entry.getKey(), this.source.getQueueId());
+    updateReplicationMarkerEdit(entry, batch.getLastWalPosition());
```

Comment:
Where do we add this marker to the system table of source cluster? I just see we add some WAL roll event to namedQueueRecord.

Reply:
Ah, read the design doc again, we do not need to persist the mark in source cluster.